packages/client/screens/MyProfile/__tests__/MyProfile.test.tsx

```diff
@@ -71,3 +71,54 @@ test('It should display my profile not found screen', async () => {
         });
     });
 });
+
+test('It should display my library after pressing playlists counter', async () => {
+    const userID = testGetFakeUserID();
+
+    db.myProfileInformation.create({
+        userID,
+        devicesCounter: 3,
+        playlistsCounter: 4,
+        followersCounter: 5,
+        followingCounter: 6,
+        userNickname: internet.userName(),
+    });
+
+    const screen = await renderApp();
+
+    expect(screen.getAllByText(/home/i).length).toBeGreaterThanOrEqual(1);
+
+    const myProfileButton = screen.getByTestId('open-my-profile-page-button');
+    expect(myProfileButton).toBeTruthy();
+
+    fireEvent.press(myProfileButton);
+
+    const playlistsCounter = await waitFor(() => {
+        expect(screen.getByTestId('my-profile-page-container')).toBeTruthy();
+
+        const devicesCounter = screen.getByText(/devices.*3/i);
+        expect(devicesCounter).toBeTruthy();
+
+        const playlistsCounter = screen.getByText(/playlists.*4/i);
+        expect(playlistsCounter).toBeTruthy();
+
+        const followersCounter = screen.getByText(/followers.*5/i);
+        expect(followersCounter).toBeTruthy();
+
+        const followingCounter = screen.getByText(/following.*6/i);
+        expect(followingCounter).toBeTruthy();
+
+        const avatar = screen.getByLabelText(/my.*avatar/i);
+        expect(avatar).toBeTruthy();
+
+        return playlistsCounter;
+    });
+
+    expect(playlistsCounter).toBeTruthy();
+
+    fireEvent.press(playlistsCounter);
+
+    await waitFor(() => {
+        expect(screen.getByTestId('library-mpe-rooms-list')).toBeTruthy();
+    });
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏This assertion is not sufficient to ensure that the library is displayed when the playlists button is pressed. Indeed, My Library screen is always displayed, as all tabs.

I would suggest to also check that My Profile screen is hidden. This is not the best indication that actually we are displaying My Library screen, but it means that the last screen as been exited.

```suggestion
    await waitFor(() => {
        expect(screen.getByTestId('library-mpe-rooms-list')).toBeTruthy();

        expect(screen.queryByTestId('my-profile-page-container')).toBeNull();
    });
```

Reply:
When I write tests, I try to always verify that tests actually break if the implementation is reverted to its previous state. Often I forget to do so, nonetheless 😆.

Reply:
Yes you're right

**💭 thought:**How could we test the following funnel:
Home -> press button that redirects to library
As the home and library would then both appears to be truthy ?

Reply:
Wow, good question!
It may be possible to use `useIsFocused` hook from React Navigation to conditionally apply a prop to `AppContainer`, and check it in tests with something like `expect(screen).toHaveProp('screen-focused', true)`?

Reply:
Yep sounds possible :open_mouth: